pkgpanda/__init__.py

```diff
@@ -442,15 +452,54 @@ def validate_username(username):
         if not re.match(username_regex, username):
             raise ValidationError("Username must begin with `dcos_` and only have a-z and underscore after that")
 
-    def add_user(self, username):
+    @staticmethod
+    def validate_group(group):
+        # Empty group is allowed.
+        if not group:
+            return
+
+        UserManagement.validate_group_name(group)
+
+        try:
+            grp.getgrnam(group)
+        except KeyError:
+            raise ValidationError("Group {} does not exist on the system".format(group))
+
+    @staticmethod
+    def validate_group_name(group_name):
+        if not group_name:
+            return
+
+        if not re.match(linux_group_regex, group_name):
+            raise ValidationError("Group {} has invalid name, must match the following regex: {}".format(
+                group_name, linux_group_regex))
+
+    @staticmethod
+    def validate_user_group(username, group_name):
+        user = pwd.getpwnam(username)
+        if not group_name:
+            return
+
+        group = grp.getgrnam(group_name)
+        if user.pw_gid != group.gr_gid:
+
+            # check if the user is the right group, but the group is not primary.
+            if username in group.gr_mem:
```

Comment:
@cmaloney squashed the commits and added check for username in other groups here ^^